bin/prepare_score_distribution.py

```diff
@@ -0,0 +1,94 @@
+import sys
+import os
+import csv
+from tqdm import tqdm
+
+import pandas as pd
+
+####################################################################################################
+
+def parse_args(args=None):
+    """Parses the command line arguments specified by the user."""
+    parser = argparse.ArgumentParser(description="Prepare prediction score distribution for plotting.")
+
+    # INPUT FILES
+    parser.add_argument("-p"     , "--predictions"              , help="Path to the predictions input file"                     , type=str   , required=True)
+    parser.add_argument("-ppo"   , "--protein-peptide-occ"      , help="Path to the protein peptide occurences input file"      , type=str   , required=True)
+    parser.add_argument("-mpo"   , "--microbiome-protein-occ"   , help="Path to the microbiome protein occurences input file"   , type=str   , required=True)
+    #parser.add_argument("-c"     , "--conditions"               , help="Path to the conditions input file"                      , type=str   , required=True)
+    #parser.add_argument("-cam"   , "--condition-allele-map"     , help="Path to the condition allele map input file"            , type=str   , required=True)
+    parser.add_argument("-a"     , "--alleles"                  , help="Path to the allele input file"                          , type=str   , required=True)
+
+    # OUTPUT FILES
+    parser.add_argument("-o"     , "--outdir"                   , help="Path to the output directory"                           , type=str   , required=True)
+
+    # PARAMETERS
+    return parser.parse_args()
+
+
+
+def main(args=None):
+    args = parse_args(args)
+
+    # Read input files
+    predictions               = pd.read_csv(args.predictions, sep='\t')
+    protein_peptide_occs      = pd.read_csv(args.protein_peptide_occ, sep='\t').drop(columns="count")
+    microbiome_protein_occs   = pd.read_csv(args.microbiome_protein_occ, sep='\t')
+    #conditions                = pd.read_csv(args.conditions, sep='\t').drop(columns="condition_name")
+    #condition_allele_map      = pd.read_csv(args.condition_allele_map, sep='\t')
+    alleles                   = pd.read_csv(args.alleles, sep='\t')
+
+    # Create output directory if it doesn't exist
+    if os.path.exists(args.outdir) and not os.path.isdir(args.outdir):
+        print("ERROR - The target path is not a directory", file = sys.stderr)
+        sys.exit(2)
+    elif not os.path.exists(args.outdir):
+        os.makedirs(args.outdir)
+
+    print("Joining input data...", file = sys.stderr, flush=True, end='')
+
+    # for each allele separately (to save mem)
+    for allele_id in alleles.allele_id:
+        print("Process allele: ", allele_id, flush=True)
+
+        data = predictions[predictions.allele_id == allele_id] \
+                .drop(columns="allele_id") \
+                .merge(protein_peptide_occs) \
+                .merge(microbiome_protein_occs) \
+                .drop(columns="protein_id") \
```

Comment:
Missing join, the current version links every allele with every microbiome, even if they didn't co-occur in the same condition.

```suggestion
                .drop(columns="protein_id") \
                .merge(conditions) \
                .merge(condition_allele_map) \
                .drop(columns="condition_id") \
```